### PR TITLE
feat(therundown): integrations tests

### PR DESCRIPTION
**Pull request by 1marcghannam**

### CODE REVIEW

File: packages/sources/therundown/test/integration/schedule.ts [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
 } from './fixtures'
 
 export function scheduleTests(): void {
+  const req: SuperTest<Test> = request('localhost:8080')
   const id = '1'
-  const req = request('localhost:8080')
 
   describe('error calls', () => {
-    describe('when unsuccessfully requesting Therundown API', () => {
+    describe('when unsuccessfully requesting Therundown API with Market Create', () => {
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/schedule.ts]: The tests below are testing the same thing:
- `error calls when unsuccessfully requesting Therundown API with Market Create should throw an exception`
- `error calls when unsuccessfully requesting Therundown API with Market Resolve should throw an exception`

The response from `mockScheduleResponseError` fixture will throw an error in `requester.ts`. It is not input-parameter dependent. Remove one of the tests and amend the test removing the market mention, e.g:
`error calls when unsuccessfully requesting Therundown API should throw an exception`

File: packages/sources/therundown/test/integration/schedule.ts [hunk 2]
```diff
@@ -38,7 +65,32 @@ export function scheduleTests(): void {
     })
 
     describe('when successfully requesting Therundown API', () => {
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/schedule.ts:42]: Both "malformed" tests should be inside the same `describe` (describe 2):
- describe1: `error calls`
  - describe 2: `when successfully requesting Therundown API`:
    - test 1: `should throw an exception if a game attribute is malformed while creating market`
    - test 2: `should throw an exception if a game attribute is malformed while resolving market`

There are many ways to organize them but given describe 2 it make sense to have them both within.

File: packages/sources/therundown/test/integration/schedule.ts [hunk 3]
```diff
@@ -64,6 +116,30 @@ export function scheduleTests(): void {
   })
 
   describe('success calls', () => {
+    describe('when successfully requesting Therundown API', () => {
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/schedule.ts:91]: Same than above, both `success calls` tests should be within `when successfully requesting Therundown API` describe.

File: packages/sources/therundown/test/integration/schedule.ts [hunk 4]
```diff
@@ -64,6 +116,30 @@ export function scheduleTests(): void {
...
+      it('should return correct game attributes', async () => {
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/schedule.ts]: This is the jest summary:
```
      success calls
        when successfully requesting Therundown API
          ✓ should return correct game attributes (7 ms)
          ✓ should return correct game attributes (9 ms)
```
Please, extend the title of both tests to help differentiating them, for instance:
```
      success calls
        when successfully requesting Therundown API
          ✓ should return correct game attributes (case create market) (7 ms)
          ✓ should return correct game attributes (case resolve market) (9 ms)
```

File: packages/sources/therundown/test/integration/schedule.ts [hunk 5]
```diff
@@ -64,6 +116,30 @@ export function scheduleTests(): void {
...
+
+        const response = await req
+          .post('/')
+          .send(data)
+          .set('Accept', '*/*')
+          .set('Content-Type', 'application/json')
+          .expect('Content-Type', /json/)
+          .expect(200)
+
+        assertSuccess({ expected: 200, actual: response.statusCode }, response.body, id)
```
Comment (vnavascues) [inline packages/sources/therundown/test/integration/schedule.ts]: Missing `expect(response.body).toMatchSnapshot()`. This test is not creating a snapshot.

Idk yet if snapshots will be used for generating the documentation output, but they are a helpful tool to make sure the EA output is the expected one (by the given input params). Make sure each successful test you add has a snapshot and use it to help yourself.

**Review by vnavascues — APPROVED** [review]
Thanks for rolling back to the previous test structure.
1 line to add (missing snapshots) and a little refactor (removing duplicated tests & describes)

File: packages/sources/therundown/test/integration/schedule.ts (hunk 1 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/schedule.ts]: Resolved

File: packages/sources/therundown/test/integration/schedule.ts (hunk 2 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/schedule.ts:42]: Resolved

File: packages/sources/therundown/test/integration/schedule.ts (hunk 3 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/schedule.ts:91]: Can you take a look again at this one? the describe(success calls) contains tests for when the api successfully returns a good response, unlike the malformed.

File: packages/sources/therundown/test/integration/schedule.ts (hunk 4 above)
Comment (1marcghannam) [reply packages/sources/therundown/test/integration/schedule.ts]: Resolved

**Pull request merged**